lib/dradis/plugins/templates.rb

```diff
@@ -40,16 +83,81 @@ def plugin_templates(args={})
 
         private
 
-        # Normally we want to copy all templates so that the user always has
-        # the latest version.
-        #
-        # However, if it's a '.template' file, the user might have edited their
-        # local copy, and we don't want to override their changes.  So only
-        # copy .template files over if the user has no copy at all (i.e. if
-        # this is the first time they've started Dradis since this template was
-        # added.)
+        def create_mapping(mapping_source)
+          destination = @rtp_id ? "rtp_#{@rtp_id}" : nil
+
+          Mapping.find_or_create_by!(
+            component: @integration_name,
+            source: mapping_source,
+            destination: destination
+          )
+        end
+
+        def create_mapping_fields(mapping, template_file)
+          template_fields = parse_template_fields(template_file)
+
+          # create a mapping_field for each field in the .template file
+          template_fields.each do |field_title, field_content|
+            # set source_field by taking the first match to the existing %% syntax
+            source_field = field_content.match(LEGACY_FIELDS_REGEX)
+            source_field = source_field ? source_field[1] : 'custom text'
```

Comment:
I don't know about this "custom text" literal, but if you're sure, go for it!

Reply:
We're also using this in the mappings form to allow for source fields that don't match any of the provided fields
ex.
```
#[Title]#
only text content
```
vs.
```
#[Title]#
{{ brakeman[issue.title] }}
```